sklearn/preprocessing/_encoders.py

```diff
@@ -114,28 +162,33 @@ def _transform(self, X, handle_unknown='error'):
 
         for i in range(n_features):
             Xi = X_list[i]
-            diff, valid_mask = _encode_check_unknown(Xi, self.categories_[i],
-                                                     return_mask=True)
 
-            if not np.all(valid_mask):
-                if handle_unknown == 'error':
-                    msg = ("Found unknown categories {0} in column {1}"
-                           " during transform".format(diff, i))
-                    raise ValueError(msg)
-                else:
-                    # Set the problematic rows to an acceptable value and
-                    # continue `The rows are marked `X_mask` and will be
-                    # removed later.
-                    X_mask[:, i] = valid_mask
-                    # cast Xi into the largest string type necessary
-                    # to handle different lengths of numpy strings
-                    if (self.categories_[i].dtype.kind in ('U', 'S')
-                            and self.categories_[i].itemsize > Xi.itemsize):
-                        Xi = Xi.astype(self.categories_[i].dtype)
+            is_category = (self.categories == 'dtypes' and
+                           Xi.dtype.name == 'category')
+            # categories without missing values do not have unknown values
```

Comment:
Similarly to my comment about `_encode`, this could also be handled in `_encode_check_unknown` in principle